singer/src/main/java/com/pinterest/singer/writer/KafkaWriter.java

```diff
@@ -328,18 +364,39 @@ public void writeLogMessages(List<LogMessage> logMessages) throws LogStreamWrite
 
       int bytesWritten = 0;
       int maxKafkaBatchWriteLatency = 0;
+      boolean anyBucketSendFailed = false;
+
       for (Future<KafkaWritingTaskResult> f : resultSet) {
         KafkaWritingTaskResult result = f.get();
         if (!result.success) {
           LOG.error("Failed to write messages to kafka topic {}", topic, result.exception);
-          throw new LogStreamWriterException("Failed to write messages to kafka");
+          anyBucketSendFailed = true;
         } else {
           bytesWritten += result.getWrittenBytesSize();
           // get the max write latency
           maxKafkaBatchWriteLatency = Math.max(maxKafkaBatchWriteLatency,
               result.getKafkaBatchWriteLatencyInMillis());
+          if (isLoggingAuditEnabledAndConfigured()) {
+            int bucketIndex = result.getPartition();
+            if (bucketIndex >= 0 && result.getRecordMetadataList().size() != buckets.get(bucketIndex).size()){
```

Comment:
could you add comment to explain why we need this check ?

Reply:
added comments in the update of the PR